packages/shared/utils/time.test.ts

```diff
@@ -4,44 +4,81 @@ import { isRangeInRegions } from "./time";
+      return ranges;
+    }
+
+    function tuplesToRanges(...tuples: any[]): TimeStampedPointRange[] {
+      const ranges: TimeStampedPointRange[] = [];
       for (let i = 0; i < tuples.length; i += 2) {
-        const beginPoint = tuples[i];
-        const endPoint = tuples[i + 1];
-        range.push({
-          begin: {
-            point: beginPoint,
-            time: parseInt(beginPoint, 10),
-          },
-          end: {
-            point: endPoint,
-            time: parseInt(endPoint, 10),
-          },
-        });
+        ranges.push(tupleToRange(tuples[i], tuples[i + 1]));
       }
-      return range;
+      return ranges;
+    }
+
+    function tupleToRange(
+      beginPoint: ExecutionPoint,
+      endPoint: ExecutionPoint
+    ): TimeStampedPointRange {
+      return createRange(beginPoint, parseInt(beginPoint, 10), endPoint, parseInt(endPoint, 10));
     }
 
     it("should accept exact matches", () => {
-      expect(isRangeInRegions("0", "10", toTimeStampedPointRange("0", "10"))).toBe(true);
+      expect(isRangeInRegions(tupleToRange("0", "10"), tuplesToRanges("0", "10"))).toBe(true);
     });
 
     it("should accept subsets", () => {
-      expect(isRangeInRegions("0", "1", toTimeStampedPointRange("0", "10"))).toBe(true);
-      expect(isRangeInRegions("1", "9", toTimeStampedPointRange("0", "10"))).toBe(true);
-      expect(isRangeInRegions("5", "10", toTimeStampedPointRange("0", "10"))).toBe(true);
+      expect(isRangeInRegions(tupleToRange("0", "1"), tuplesToRanges("0", "10"))).toBe(true);
+      expect(isRangeInRegions(tupleToRange("1", "9"), tuplesToRanges("0", "10"))).toBe(true);
+      expect(isRangeInRegions(tupleToRange("5", "10"), tuplesToRanges("0", "10"))).toBe(true);
     });
 
     it("should reject partial overlaps", () => {
-      expect(isRangeInRegions("0", "5", toTimeStampedPointRange("5", "10"))).toBe(false);
-      expect(isRangeInRegions("2", "7", toTimeStampedPointRange("5", "10"))).toBe(false);
-      expect(isRangeInRegions("0", "15", toTimeStampedPointRange("5", "10"))).toBe(false);
-      expect(isRangeInRegions("7", "12", toTimeStampedPointRange("5", "10"))).toBe(false);
+      expect(isRangeInRegions(tupleToRange("0", "5"), tuplesToRanges("5", "10"))).toBe(false);
+      expect(isRangeInRegions(tupleToRange("2", "7"), tuplesToRanges("5", "10"))).toBe(false);
+      expect(isRangeInRegions(tupleToRange("0", "15"), tuplesToRanges("5", "10"))).toBe(false);
+      expect(isRangeInRegions(tupleToRange("7", "12"), tuplesToRanges("5", "10"))).toBe(false);
     });
 
     it("should reject non-contiguous overlaps", () => {
-      expect(isRangeInRegions("2", "5", toTimeStampedPointRange("0", "3", "4", "8"))).toBe(false);
+      expect(isRangeInRegions(tupleToRange("2", "5"), tuplesToRanges("0", "3", "4", "8"))).toBe(
+        false
+      );
+    });
+
+    // Imitate scenario where in-memory time-to-points mapping is too coarse
+    // to correctly differentiate between different time ranges.
```

Comment:
I think I'm having a hard time envisioning this case 🤔

Reply:
I left a comment this morning (on one of your Replays IIR) where this seemed to be happening.